pyodide-build/pyodide_build/_py_compile.py

```diff
@@ -190,20 +195,131 @@ def _py_compile_wheel(
 
     Returns
     -------
-    wheel_path_out
-        processed wheel with .pyc files.
+    path
+        path to processed archive with .pyc files. Or None if the file was not py-compiled
+        (e.g. if it's a zip with no .py files)
+    """
+    if input_path.suffix not in [".whl", ".zip"]:
+        raise ValueError(f"Error: only .whl files are supported, got {input_path.name}")
+
+    if not input_path.exists():
+        raise FileNotFoundError(f"{input_path} does not exist!")
+
+    name_out = _get_py_compiled_archive_name(input_path)
+    if name_out is None:
+        return None
+    path_out = input_path.parent / name_out
+
+    _compile(input_path, path_out, keep=keep, verbose=verbose)
+
+    return path_out
 
 
+def _get_py_compiled_archive_name(path: Path) -> str | None:
+    """Return the name of the py-compiled wheel or zip file
+
+    Returns None if the file should not be py-compiled.
+
+    Examples
+    --------
+    >>> _get_py_compiled_archive_name(Path("snowballstemmer-2.2.0-py2.py3-none-any.whl"))
+    'snowballstemmer-2.2.0-cp311-none-any.whl'
+    >>> _get_py_compiled_archive_name(Path("test-1.0.0.zip"))
+    """
+    # TODO: fix py-compilation of the following packages
+    if path.name.startswith(("RobotRaconteur", "astropy-", "opencv_python-")):
```

Comment:
What's the problem with these packages?

Reply:
- RobotRaconteur has some file with invalid syntax
 - opencv-python explicitly requires the `config.py` file to exist, so it shouldn't be compiled but then it breaks our test assumtion of all or nothing for .pyc
 - astropy also py-compiles fine but fails with an import error, don't remember why now.
 
 Overall because this PR is already quite large I would rather fix those separately. Will open an issue.